crates/ruff_db/src/vfs.rs

```diff
@@ -259,6 +259,40 @@ impl VfsFile {
 
         db.vfs().read(db, path)
     }
+
+    /// Refreshes the file metadata by querying the file system if needed.
+    /// TODO: The API should instead take all observed changes from the file system directly
+    ///   and then apply the VfsFile status accordingly. But for now, this is sufficient.
+    pub fn touch_path(db: &mut dyn Db, path: &VfsPath) {
+        Self::touch_impl(db, path, None);
+    }
+
+    pub fn touch(self, db: &mut dyn Db) {
+        let path = self.path(db).clone();
+        Self::touch_impl(db, &path, Some(self));
+    }
+
+    fn touch_impl(db: &mut dyn Db, path: &VfsPath, file: Option<VfsFile>) {
```

Comment:
(The name of this wasn't immediately clear to me when I skimmed by it on the first read, until I looked at the surrounding code on the second read)

```suggestion
    /// Private method providing the implementation for `touch_path` and `touch`
    fn touch_impl(db: &mut dyn Db, path: &VfsPath, file: Option<VfsFile>) {
```